# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderImageHost.cpp
# Commit: de6ddf5eb2a7
# Full Hash: de6ddf5eb2a7bd3d8a2f58636760e0aa16c584b9
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 773e7b31ac6e -r de6ddf5eb2a7 gfx/layers/wr/WebRenderImageHost.cpp
--- a/gfx/layers/wr/WebRenderImageHost.cpp	Tue Dec 05 09:37:03 2023 +0000
+++ b/gfx/layers/wr/WebRenderImageHost.cpp	Tue Dec 05 09:37:03 2023 +0000
@@ -143,6 +143,7 @@
       // Clear when RemoteTextureOwner is different.
       mPendingRemoteTextureWrappers.clear();
       mWaitingReadyCallback = false;
+      mWaitForRemoteTextureOwner = true;
     }
   }
 
@@ -222,7 +223,8 @@
 
     std::function<void(const RemoteTextureInfo&)> function;
     RemoteTextureMap::Get()->GetRemoteTextureForDisplayList(
-        wrapper, std::move(function));
+        wrapper, std::move(function), mWaitForRemoteTextureOwner);
+    mWaitForRemoteTextureOwner = false;
   }
 
   if (!texture ||